packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -352,6 +352,21 @@ exports.createContentTypeNodes = ({
           childrenNodes.push(jsonNode)
           entryItemFields[`${entryItemFieldKey}___NODE`] = jsonNode.id
 
+          delete entryItemFields[entryItemFieldKey]
+        } else if (fieldType === `Object` && _.isArray(entryItemFields[entryItemFieldKey])) {
+          entryItemFields[`${entryItemFieldKey}___NODE`] = []
+
+          entryItemFields[entryItemFieldKey].forEach((obj, i) => {
+            const jsonNode = prepareJSONNode(
+              entryNode,
+              entryItemFieldKey,
+              obj
+            )
```

Comment:
Does this work properly? It seems for me that it will create array of nodes all with same ID which means last node would overwrite all previous nodes

Reply:
@pieh it does, here an [output demo](https://prnt.sc/jqnc5l)